### PR TITLE
Merge PR status update commands with pr update commands

### PR DESCRIPTION
**Pull request by gauravsaralMs**
Please make sure the code is following contribution guidelines in CONTRIBUTING.md

 - [ ] : This PR has a corresponding issue open in the Repository.
 - [ ] : Approach is signed off on the issue.

Fixes: https://github.com/Azure/azure-devops-cli-extension/issues/604

image

image

### CODE REVIEW

File: tests/test_reposPrCommandsReviewersTest.py [hunk 1]
```diff
@@ -65,15 +65,14 @@ def test_pull_request_createUpdateVoteListAbandonReactivateCompleteReviewers(sel
             show_pr_description = show_pr_output["description"]
             assert show_pr_title == pr_title
             assert show_pr_description == update_pr_description
-            
+
             #Abandon PR
-            abandon_pr_command = 'az repos pr abandon --id ' + create_pr_id + ' --detect Off --output json'
+            abandon_pr_command = 'az repos pr update --id ' + create_pr_id + ' --detect Off --output json --status abandoned --debug'
```
Comment (ishitam8) [inline tests/test_reposPrCommandsReviewersTest.py]: We can remove debug here.

File: azure-devops/azext_devops/dev/repos/pull_request.py [hunk 2]
```diff
@@ -306,43 +308,18 @@ def update_pull_request(id, title=None, description=None, auto_complete=None,  #
             pr.auto_complete_set_by = IdentityRef(id=EMPTY_UUID)
     if draft is not None:
         pr.is_draft = draft
+    if status is not None:
+        pr.status = status
+        if status == 'completed':
+            pr.last_merge_source_commit = existing_pr.last_merge_source_commit
+            pr.completion_options = existing_pr.completion_options
```
Comment (atbagga) [inline azure-devops/azext_devops/dev/repos/pull_request.py]: This should be pr.completion options now. If I update the completion options and status in the same update command you will end up overwriting the completion options after setting them in lines 289-303.

Comment (gauravsaralMs) [reply azure-devops/azext_devops/dev/repos/pull_request.py]: nice catch

File: tests/test_reposPrCommandsReviewersTest.py (hunk 1 above)
Comment (gauravsaralMs) [reply tests/test_reposPrCommandsReviewersTest.py]: done

File: azure-devops/azext_devops/dev/repos/pull_request.py (hunk 2 above)
Comment (atbagga) [reply azure-devops/azext_devops/dev/repos/pull_request.py]: that is being set selectively so you will need to handle appropriately. 

When the update is to set the status as completed. PR code on server needs the completion options to be sent with the request.

**Pull request merged**